### PR TITLE
Disable payment gateway when not in test mode and not using https or ssl checkout enforcement

### LINKED ISSUE #2374: Set the gateway to unavailable if in live mode and not on HTTPS
We currently use Stripe elements which will disable the payments if not using HTTPS in live mode.

Would be good to add an extra layer of security, keep it consistent with other payment extensions, and clearer to the admin by doing this check ourselves.

We should modify the `is_available` method to return false when the gateway mode is not test and `'yes' === get_option( 'woocommerce_force_ssl_checkout' ) || wc_site_is_https()` evaluates to false.  

The existing `WC_Payments_Notes_Set_Https_For_Checkout` note should handle notifying the merchant of the HTTPS requirement like it does now, but maybe we should also add a task to the account overview page?

### PR DESCRIPTION
**Pull request by daquinons**
Fixes #2374 

#### Changes proposed in this Pull Request
PHP side:
- `is_available` method now checks for not being in test mode, using `https` or having the ssl in checkout enforced.
-  Passes `woocommerce_force_ssl_checkout` to React in order to use it in the Task list.

JS side:
- Adds a new Task if not in Test mode, not using https or not having woocommerce_force_ssl_checkout enabled.
-  The new Task uses `actionLabel` and `action`. The old tasks will need to fix `action` usage (see #2479).

image

#### Testing instructions

* Disable Test mode
* Disable SSL checkout enforcement here http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=advanced
* Go to Payments Overview and see the task active
* Try to buy something
* Go to checkout
* See payment gateway disabled in test mode and without https or ssl checkout enforcement:  `Sorry, it seems that there are no available payment methods for your state. Please contact us if you require assistance or wish to make alternate arrangements.`
* Enable SSL checkout enforcement
* Go back to Payments, see task not showing up

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/class-wc-payment-gateway-wcpay.php [hunk 1]
```diff
@@ -497,6 +497,11 @@ public function is_in_test_mode() {
 	 * @return bool Whether the gateway is enabled and ready to accept payments.
 	 */
 	public function is_available() {
+		// Disable the gateway if using live mode without HTTPS set up.
+		if ( ! $this->is_in_test_mode() && ( ! wc_site_is_https() || 'yes' !== get_option( 'woocommerce_force_ssl_checkout' ) ) ) {
```
Comment (luizreis) [inline includes/class-wc-payment-gateway-wcpay.php:501]: There's a [`wc_checkout_is_https` function](https://github.com/woocommerce/woocommerce/blob/b19500728b4b292562afb65eb3a0c0f50d5859de/includes/wc-conditional-functions.php#L408), which looks similar to this but includes a couple of more checks. Would it make sense to use it here?

File: includes/admin/class-wc-payments-admin.php [hunk 2]
```diff
@@ -279,6 +279,7 @@ public function register_payments_scripts() {
 				'isUpeSettingsPreviewEnabled' => WC_Payments_Features::is_upe_settings_preview_enabled(),
 				'isUpeEnabled'                => WC_Payments_Features::is_upe_enabled(),
 			],
+			'isSSLCheckoutEnforced'  => get_option( 'woocommerce_force_ssl_checkout', 'no' ),
```
Comment (luizreis) [inline includes/admin/class-wc-payments-admin.php:282]: What do you think of passing the same value used in the `is_available` check here? We could extract the check to a new method and use it in both places. That way, the JS and PHP logic would be shared, the check to display the notice would be simpler, and we wouldn't need `isSiteUsingHttps`.

File: includes/class-wc-payment-gateway-wcpay.php (hunk 1 above)
Comment (daquinons) [reply includes/class-wc-payment-gateway-wcpay.php]: This is GOLD 🥇 Thanks for pointing it out, it's actually removing all the logic 👍

File: includes/admin/class-wc-payments-admin.php (hunk 2 above)
Comment (daquinons) [reply includes/admin/class-wc-payments-admin.php]: Great idea! I've changed it for `isGatewayAvailable` and passed it down to display the task. Also removed the unused `isSiteUsingHttps` utility.

**Review by luizreis — APPROVED** [review]
LGTM! Thanks for making the adjustments :shipit:

**Pull request merged**